tests/e2e-pw/tests/product-gallery/inner-blocks/product-gallery-thumbnails/product-gallery-thumbnails.block_theme.spec.ts

```diff
@@ -0,0 +1,374 @@
+			postId: `woocommerce/woocommerce//${ blockData.slug }`,
+			postType: 'wp_template',
+		} );
+		await editorUtils.enterEditMode();
+	} );
+
+	test.afterEach( async ( { requestUtils } ) => {
+		await requestUtils.deleteAllTemplates( 'wp_template' );
+		await requestUtils.deleteAllTemplates( 'wp_template_part' );
+	} );
+
+	test( 'Renders Product Gallery Thumbnails block on the editor side', async ( {
+		editorUtils,
+		editor,
+	} ) => {
+		await editor.insertBlock( {
+			name: 'woocommerce/product-gallery',
+		} );
+
+		const block = await editorUtils.getBlockByName( blockData.name );
+
+		await expect( block ).toBeVisible();
+	} );
+
+	test( 'Renders Product Gallery Thumbnails block on the frontend side', async ( {
+		editorUtils,
+		editor,
+	} ) => {
+		await editor.insertBlock( {
+			name: 'woocommerce/product-gallery',
+		} );
+
+		const block = await editorUtils.getBlockByName( blockData.name );
+
+		await expect( block ).toBeVisible();
+	} );
+
+	test.describe( `${ blockData.name } Settings`, () => {
+		test( 'Hide correctly the thumbnails', async ( {
+			page,
+			editor,
+			editorUtils,
+			admin,
+		} ) => {
+			await addBlock( admin, editor, editorUtils );
+			await (
+				await editorUtils.getBlockByName( blockData.name )
+			 ).click();
+			await editor.openDocumentSettingsSidebar();
+			await page
+				.locator( blockData.selectors.editor.noThumbnailsOption )
+				.click();
+
+			const isVisible = await page
+				.locator( blockData.selectors.editor.thumbnails )
+				.isVisible();
+
+			expect( isVisible ).toBe( false );
+
+			await editor.saveSiteEditorEntities();
+
+			await page.goto( blockData.productPage, {
+				waitUntil: 'networkidle',
+			} );
+		} );
+
+		// We can test the left position of thumbnails by cross-checking:
+		// - The Gallery block has the classes "is-layout-flex" and "is-nowrap".
+		// - The Thumbnails block has a lower index than the Large Image block.
+		test( 'Position thumbnails on the left of the large image', async ( {
+			page,
+			editor,
+			editorUtils,
+			frontendUtils,
+		} ) => {
+			// Currently we are adding the block under the legacy Product Image Gallery block, but in the future we have to add replace the product gallery block with this block.
+			const parentBlock = await editorUtils.getBlockByName(
+				'woocommerce/product-image-gallery'
+			);
+			const clientId =
+				( await parentBlock.getAttribute( 'data-block' ) ) ?? '';
+			const parentClientId =
+				( await editorUtils.getBlockRootClientId( clientId ) ) ?? '';
+
+			await editor.selectBlocks( parentBlock );
+			await editorUtils.insertBlock(
+				{ name: 'woocommerce/product-gallery' },
+				undefined,
+				parentClientId
+			);
+			await (
+				await editorUtils.getBlockByName( blockData.name )
+			 ).click();
+
+			await editor.openDocumentSettingsSidebar();
+			await page
+				.locator(
+					blockData.selectors.editor.leftPositionThumbnailsOption
+				)
+				.click();
+
+			await page.waitForTimeout( 500 );
```

Comment:
Have you considered to use Playwright's built-in methods like [`waitForSelector`](https://playwright.dev/docs/api/class-elementhandle#element-handle-wait-for-selector) here instead of `waitForTimeout`?  Waiting for arbitrary timeouts can make tests flaky and slow.

Reply:
Thanks. Yes, I am aware that `waitForSelector` is preferable, but in this case, I don't believe there really is a good selector to be waiting for. I'm happy to discuss during our pairing session :)